src/react-components/styles/theme.js

```diff
@@ -39,14 +46,18 @@ function useDarkMode() {
     // https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener
     // We may remove this workaround when no one will use Safari 13 or before.
     if (darkmodeQuery.addEventListener) {
-      darkmodeQuery.addEventListener("change", event => {
-        setDarkMode(event.matches);
-      });
+      darkmodeQuery.addEventListener("change", changeListener);
     } else {
-      darkmodeQuery.addListener(event => {
-        setDarkMode(event.matches);
-      });
+      darkmodeQuery.addListener(changeListener);
     }
+
+    return () => {
+      if (darkmodeQuery.removeEventListener) {
+        darkmodeQuery.removeEventListener("change", changeListener);
+      } else {
+        darkmodeQuery.removeListener(changeListener);
+      }
+    };
   }, []);
```

Comment:
Technically I think you need to add `changeListener` as an effect dependency now (though in practice I suspect the `setDarkMode` reference never actually changes.)
```suggestion
  }, [changeListener]);
```

Reply:
Added